src/plugins/kernel_install/utils.sh

```diff
@@ -61,67 +109,174 @@ function reboot_machine()
 {
   local reboot="$1"
   local local="$2"
+  local flag="$3"
 
-  if [[ "$local" == 'local' ]]; then
-    sudo_cmd="sudo -E"
-  fi
+  [[ "$local" == 'local' ]] && sudo_cmd="sudo -E"
 
   if [[ "$reboot" == "1" ]]; then
     cmd="$sudo_cmd reboot"
     cmd_manager "$flag" "$cmd"
   fi
 }
 
+function install_modules()
+{
+  local module_target="$1"
+  local flag="$2"
+  local ret
+
+  if [[ -z "$module_target" ]]; then
+    module_target=*.tar
+  fi
+
+  cmd_manager "$flag" "tar -C /lib/modules -xf $module_target"
+  ret="$?"
+
+  if [[ "$ret" != 0 ]]; then
+    echo "Warning: Couldn't extract module archive."
+  fi
+}
+
+# Update boot loader API
+function update_boot_loader()
+{
+  local name="$1"
+  local distro="$2"
+  local target="$3"
+  local cmd_init="$4"
+  local setup_grub="$5"
+  local grub_install="$6"
+  local flag="$7"
+
+  if [[ "$target" == 'local' ]]; then
+    sudo_cmd="sudo -E"
+  fi
+
+  cmd_grub="$sudo_cmd grub-mkconfig -o /boot/grub/grub.cfg"
+
+  # Update grub
+  if [[ "$target" == 'vm' ]] ; then
+    vm_update_boot_loader "$name" "$distro" "$cmd_grub" "$cmd_init" "$setup_grub" "$grub_install" "$flag"
+  else
+    cmd_manager "$flag" "$cmd_grub"
+  fi
+}
+
+# After configuring the handle (adding a disk image in write mode), the
+# guestfish performes the followed steps: (1) mount image;
+# (2) dracut (updates kernel images list); (3) create a dummy device.map
+# that tells Grub to look for /dev/sda; (4) install and update grub.
+#
+# Note: The virtual machine must be shut down and umounted before you use this
+# command, and disk images must not be edited concurrently.
+#
+# @name Kernel name for the deploy
+# @cmd_grub Command to update grub
+# mkinitcpio -g /boot/initramfs-linux.img -k /boot/vmlinuz-linux
+function vm_update_boot_loader()
+{
+  local name="$1"
+  local distro="$2"
+  local cmd_grub="$3"
+  local cmd_init="$4"
+  local setup_grub="$5"
+  local grub_install="$6"
+  local flag="$7"
+  local cmd=''
+  # We assume Debian as a default option
+  local mount_root=": mount /dev/sda1 /"
+  local mkdir_grub=": mkdir-p /boot/grub"
```

Comment:
I checked that lines 188 and 203 switched places. If we assume Debian as a default option, we will use the variable mkdir_init as defined in line 203. In the arch case, we use the mkdir_grub.